code/modules/antagonists/heretic/structures/mawed_crucible.dm

```diff
@@ -0,0 +1,293 @@
+			return ITEM_INTERACT_COMPLETE
+		var/obj/item/reagent_containers/glass/beaker/eldritch/to_fill = tool
+		if(to_fill.reagents.total_volume >= to_fill.reagents.maximum_volume)
+			to_chat(user, "<span class='hierophant_warning'>Your flask is full!</span>")
+			return ITEM_INTERACT_COMPLETE
+		to_fill.reagents.add_reagent("eldritch", 50)
+		do_item_attack_animation(src, used_item = tool)
+		current_mass--
+		to_chat(user, "<span class='hierophant'>Your flask has been refilled.</span>")
+		return ITEM_INTERACT_COMPLETE
+
+/obj/structure/eldritch_crucible/attack_hand(mob/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+
+	if(!isliving(user))
+		return
+
+	if(!IS_HERETIC_OR_MONSTER(user))
+		if(iscarbon(user))
+			bite_the_hand(user)
+		return TRUE
+
+	if(in_use)
+		to_chat(user, "<span class='hierophant_warning'>We are in use!</span>")
+		return TRUE
+
+	if(current_mass < max_mass)
+		to_chat(user, "<span class='hierophant_warning'>We are not full enough!</span>")
+		return TRUE
+
+	INVOKE_ASYNC(src, PROC_REF(show_radial), user)
+	return TRUE
+
+/*
+ * Wrapper for show_radial() to ensure in_use is enabled and disabled correctly.
+ */
+/obj/structure/eldritch_crucible/proc/show_radial(mob/living/user)
+	in_use = TRUE
+	create_potion(user)
+	in_use = FALSE
+
+/*
+ * Shows the user of radial of possible potions,
+ * and create the potion they chose.
+ */
+/obj/structure/eldritch_crucible/proc/create_potion(mob/living/user)
+
+	// Assoc list of [name] to [image] for the radial
+	var/static/list/choices = list()
+	// Assoc list of [name] to [path] for after the radial, to spawn it
+	var/static/list/names_to_path = list()
+	if(!choices.len || !names_to_path.len)
+		for(var/obj/item/eldritch_potion/potion as anything in subtypesof(/obj/item/eldritch_potion))
+			names_to_path[initial(potion.name)] = potion
+			choices[initial(potion.name)] = image(icon = initial(potion.icon), icon_state = initial(potion.icon_state))
+
+	var/picked_choice = show_radial_menu(
+		user,
+		src,
+		choices,
+		require_near = TRUE,
+		)
+
+	if(isnull(picked_choice))
+		return
+
+	var/spawned_type = names_to_path[picked_choice]
+	if(!ispath(spawned_type, /obj/item/eldritch_potion))
+		CRASH("[type] attempted to create a potion that wasn't an eldritch potion! (got: [spawned_type])")
+
+	var/obj/item/spawned_pot = new spawned_type(drop_location())
+
+	playsound(src, 'sound/effects/desecration/desecration-02.ogg', 75, TRUE)
+	visible_message("<span class='notice'>[src]'s shining liquid drains into a flask, creating a [spawned_pot.name]!</span>")
+
+	current_mass = 0
+	update_appearance(UPDATE_ICON_STATE)
+
+/*
+ * "Bites the hand that feeds it", except more literally.
+ * Called when a non-heretic interacts with the crucible,
+ * causing them to lose their active hand to it.
+ */
+/obj/structure/eldritch_crucible/proc/bite_the_hand(mob/living/carbon/user)
+	var/obj/item/organ/external/affecting = user.get_organ("[user.hand ? "l" : "r" ]_hand")
```

Comment:
We have to have a better way to do this. Please consider making a helper out of this

Reply:
I think this was runtiming last night so yes we do need a better way for this